server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java

```diff
@@ -15,7 +15,7 @@
  * Nodes that do not support features won't have this feature in its feature set,
  * so this can be checked without needing to look at the node version.
  */
-public class FeaturesSupportedSpecification implements FeatureSpecification {
+public class FeaturesFeatures implements FeatureSpecification {
```

Comment:
This looks funny but I don't think there is a better alternative and it does stick to the schema after all. Hope no one will rename it thinking is a typo :)

Reply:
I've added a javadoc highlighting the recursiveness of the definition

Reply:
Even with the javadoc, this name is incredibly confusing. How about FeatureDetection or FeatureInfrastructure?

Reply:
I've changed to `FeatureInfrastructureFeatures`